### DISCUSSION TITLE
What does it mean that "an empty run request is valid"?

### DISCUSSION THREAD

**Discussion by agluszak** [build-server-protocol/build-server-protocol#636] [Q&A] [answered]
See https://build-server-protocol.github.io/docs/specification#buildtargetrun-request

> Note that an empty run request is valid. Run will be executed in the target as specified in the build tool.

This is very vague. What does it mean that a run request is empty? And what does it mean that "run will be executed in the target"? And what does it mean "as specified in the build tool"?

**Comment by adpi2** [comment, marked as answer]
> What does it mean that a run request is empty?

My guess is an empty run request only contains the `target` param.

> And what does it mean that "run will be executed in the target"?

It means the build server will run the program defined in this target, if there is one.

> And what does it mean "as specified in the build tool"?

The build tool should know what are the default arguments, environment variables, and working directory to run the target.
